src/contracts/pods/EigenPod.sol

```diff
@@ -428,6 +450,104 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+        // initializes the  "timestampProvenUntil" for batched zk partial withdrawals to the GENESIS_TIME
+        if(timestampProvenUntil == 0){
+            timestampProvenUntil = GENESIS_TIME;
+        }
+
+        requestNonce++;
+        require(endTimestamp > timestampProvenUntil, "EigenPod.submitPartialWithdrawalsBatchForVerification: endTimestamp must be greater than timestampProvenUntil");
+        
+        //the proofs are made [startTimestamp, endTimestamp), so the next startTimestamp = previous endTimestamp     
+        uint64 startTimestamp = timestampProvenUntil;
+        
+        //record the partial withdrawal proof request
+       _partialWithdrawalProofRequests[requestNonce] = PartialWithdrawalProofRequest({
+            startTimestamp: startTimestamp,
+            endTimestamp: endTimestamp,
+            recipient: recipient,
+            status: REQUEST_STATUS.PENDING
+       });
+
+
+
+        emit PartialWithdrawalProofRequested(startTimestamp, endTimestamp, requestNonce);
+
+        //NOTE: it is not clear what callback data needs to passed here since succinct is going to make multiple callbacks for each request.
+        //These are the inputs to the callback function for this specific proof request.
+        bytes memory callBackData = abi.encodeWithSelector(EigenPod.handleCallback.selector, requestNonce, oracleTimestamp, _timestampToSlot(endTimestamp));
+
+
+        eigenPodManager.requestProofViaSuccinctGateway{value: msg.value}(
+            RANGE_SPLITTER_FUNCTION_ID, 
+            _timestampToSlot(timestampProvenUntil),
+            _timestampToSlot(endTimestamp),
+            address(this),
+            callBackData,
+            callbackGasLimit,
+            podOwner
+        );
+    }
+
+    /// @notice The callback function for the ZK proof fulfiller.
+    function handleCallback(uint256 nonce, uint64 oracleTimestamp, uint64 startSlot, uint64 endSlot) external onlySuccinctGateway() {
+        PartialWithdrawalProofRequest memory request = _partialWithdrawalProofRequests[nonce];
+
+        require(request.status == REQUEST_STATUS.PENDING, "EigenPod.handleCallback: request nonce is either cancelled or fulfilled");
+        require(nonce == lastRequestNonceProven + 1, "EigenPod.handleCallback: must callback in order");
+        require(startSlot < endSlot, "EigenPod.handleCallback: invalid start and end slot values");
+        require(_slotToTimestamp(endSlot) <= request.endTimestamp, "EigenPod.handleCallback: endSlot must be less than the request's endTimestamp");
+        require(_slotToTimestamp(startSlot) == timestampProvenUntil, "EigenPod.handleCallback: startSlot must be equal to the timestampProvenUntil");
+
+        bytes32 beaconBlockRoot = eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp);
+        bytes memory output = eigenPodManager.confirmProofVerification(WITHDRAWAL_FUNCTION_ID, abi.encodePacked(beaconBlockRoot, address(this), startSlot, endSlot));
```

Comment:
why have this level of indirection through the eigenpod manager instead of directly calling the succinct gateway here?